### PR TITLE
Improve search for another partner with same BSN

### LINKED ISSUE #41: l10n_nl_bsn, improve check: BSN unique
One minor bug: When having a valid BSN number for a partner, then changing it to an invalid number, save and get the correct warning, finally change it back to the original valid number, you will get a "Another person (Administrator) has the same BSN [...]" warning.

### PR DESCRIPTION
**Pull request by astirpe**
Fixes #41

### CODE REVIEW

File: l10n_nl_bsn/models/res_partner.py [hunk 1]
```diff
@@ -37,16 +37,18 @@ def onchange_bsn_number(self):
...
                 # search for another partner with the same BSN
-                args = [('bsn_number', '=', partner.bsn_number)]
+                args = [('bsn_number', '=', partner.bsn_number),
+                        ('name', '!=', partner.name)]
                 # refine search in case of multicompany setting
                 if partner.company_id:
                     args += [('company_id', '=', partner.company_id.id)]
-                partners = self.search(args, limit=1)
+                other_partner = self.search(args, limit=1)
                 # is another partner exists, display a warning
```
Comment (pankk) [inline l10n_nl_bsn/models/res_partner.py]: Typo, 'is' should be 'if' :)

**Comment by pankk** [comment]
:+1:  (code review only)

**Comment by hbrunn** [comment]
can't you search for the partner's id in the case it's available?

**Comment by astirpe** [comment]
@hbrunn I've chosen this solution for 2 reasons, one is technical and the other is functional.

**Technical**: searching in an onchange method, I cannot read my current record id, because _self.id_ is always _NewId_. This occurs also in case the record was already stored in the database.

**Functional**: let's say, for any reason, you have some duplicated partners (individuals with the same name) and you want to keep them as they are. In this case, there is no reason to raise a warning, saying that another person has the same BSN. In facts, the same duplicated partners should get the same BSN.

**Comment by hbrunn** [comment]
@astirpe thanks, makes sense. :+1:

**Comment by StefanRijnhart** [comment]
:+1:

**Pull request merged**